src/engine/resolvers.rs

```diff
@@ -80,11 +82,60 @@ where
         }
     }
 
+    /// Returns a neo4j database driver from the pool.
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Error`] variant [`DatabaseMismatch`] if the feature `neo4j`
+    /// is not enabled.
+    ///
+    /// Returns an [`Error]` variant [`Neo4jPoolGetConnectionFailed`] if a new
+    /// connection to the database cannot be retrieved from the pool.
+    ///
+    /// [`Error`]: ../../error/enum.Error.html
+    /// [`DatabaseMismatch`]: ../../error/enum.Error.html#variant.DatabaseMismatch
+    /// [`Neo4jPoolGetConnectionFailed`]: ../../error/enum.Error.html#variant.Neo4jPoolGetConnectionFailed
+    ///
+    /// # Examples
+    ///
+    /// ```rust, no_run
+    /// # use tokio::runtime::Runtime;
+    /// # use warpgrapher::engine::resolvers::{ResolverFacade, ExecutionResult};
+    ///
+    /// fn custom_resolve(facade: ResolverFacade<(), ()>) -> ExecutionResult {
+    ///     let mut rt = Runtime::new().unwrap();
+    ///     rt.block_on(async {
+    ///
+    ///         let neo4j_client = facade.db_as_neo4j().await.unwrap();
+    ///         // use neo4j client
+    ///
+    ///     });
+    ///     facade.resolve_null()
+    /// }
+    /// ```
+    #[cfg(feature = "neo4j")]
+    pub async fn db_as_neo4j(
```

Comment:
Would it make sense to put this method on the DatabasePool itself?  So the call would be `resolver.pool().as_neo4j()`. Pool just pulls the databasepool out of self.executor.context().  It's up to the DatabasePool to know how to deliver itself as a neo4j client.

Also, from this point forward, we should maintain feature parity between Cosmos and Neo4J.  If we need this convenience method for neo4j, shouldn't we also have it for Cosmos?

Reply:
Fixed.